src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/SystemFonts.cs

```diff
@@ -1023,6 +1023,22 @@ private static double ConvertFontHeight(int height)
             }
         }
 
+        private static double ConvertFontHeight(double height)
+        {
+            int dpi = SystemParameters.Dpi;
+
+            if (dpi != 0)
+            {
+                return (double)(Math.Abs(height) * 96 / dpi);
```

Comment:
The cast here does nothing. Same as it does nothing in the original function you've copied.

The original function returns `double` but the calculations are done in `int`, so it is truncated and then returned as double. Yours does the computations in double.

Reply:
Removed this

Reply:
Yeah, looks better now.

Though not for this PR, but I wonder about the original intention, the truncation is obviously not ideal for scaling but then again, it's been there since the beginning (even for NetFX).